app/src/main/java/com/github/dedis/student20_pop/model/Lao.java

```diff
@@ -1,37 +1,173 @@
+        this.attestation = name + time + organizer;
     }
 
     public String getName() {
         return name;
     }
 
+    /**
+     *
+     * @return creation time of the LAO as Unix Timestamp, can't be modified
+     */
+    public long getTime() {
+        return time;
+    }
+
+    /**
+     *
+     * @return ID of the LAO, can't be modified
+     */
+    public String getId() {
+        return id;
+    }
+
+    /**
+     *
+     * @return public key of the organizer, can't be modified
+     */
+    public String getOrganizer() {
+        return organizer;
+    }
+
+    /**
+     *
+     * @return list of public keys where each public key belongs to one witness
+     */
+    public List<String> getWitnesses() {
+        return witnesses;
+    }
+
+    /**
+     *
+     * @return list of public keys where each public key belongs to one member
+     */
+    public List<String> getMembers() {
+        return members;
+    }
+
+    /**
+     *
+     * @return list of public keys where each public key belongs to an event
+     */
+    public List<String> getEvents() {
+        return events;
+    }
+
+    /**
+     *
+     * @return signature by the organizer
+     */
+    public String getAttestation() {
+        return attestation;
+    }
+
+    /**
+     *
+     * @param name new name for the LAO, can be empty
+     * @throws IllegalArgumentException if the name is null
+     */
+    public void setName(String name) {
+        if(name == null) {
+            throw new IllegalArgumentException("Trying to set null as the name of the LAO");
+        }
+        this.name = name;
+    }
+
+    /**
+     *
+     * @param witnesses list of public keys of witnesses, can be empty
+     * @throws IllegalArgumentException if the list is null or at least one public key is null
+     */
+    public void setWitnesses(List<String> witnesses) {
+        if(witnesses == null || witnesses.contains(null)) {
+            throw new IllegalArgumentException("Trying to add a null witness to the LAO " + name);
+        }
+        this.witnesses = witnesses;
+    }
+
+    /**
+     *
+     * @param members list of public keys of members, can be empty
+     * @throws IllegalArgumentException if the list is null or at least one public key is null
+     */
+    public void setMembers(List<String> members) {
+        if(members == null || members.contains(null)) {
+            throw new IllegalArgumentException("Trying to add a null member to the LAO " + name);
+        }
+        this.members = members;
+    }
+
+    /**
+     *
+     * @param events list of public keys of events, can be empty
+     * @throws IllegalArgumentException if the list is null or at least one public key is null
+     */
+    public void setEvents(List<String> events) {
+        if(events == null || events.contains(null)) {
+            throw new IllegalArgumentException("Trying to add a null event to the LAO " + name);
+        }
+        this.events = events;
+    }
+
```

Comment:
Would a addEvent method also be useful?

Reply:
The method setEvents can be used for this